qcodes/tests/dataset/test_plotting.py

```diff
@@ -83,3 +90,62 @@ def test_rescaled_ticks_and_units(scale, unit,
     # also test the fact that "{:g}" is used in ticks formatter function
     assert '2.12346' == ticks_formatter(2.123456789 / (10 ** (-scale)))
 
+
+def test_plot_by_id_line_and_heatmap(experiment):
+    """
+    Test that line plots and heatmaps can be plotted together
+    """
+    inst = DummyInstrument('dummy', gates=['s1', 'm1', 's2', 'm2'])
+
+    inst.m1.get = lambda: np.random.randn()
+    inst.m2.get = lambda: np.random.randint(0, 5)
+
+    meas = Measurement()
+    meas.register_parameter(inst.s1)
+    meas.register_parameter(inst.s2)
+    meas.register_parameter(inst.m2, setpoints=(inst.s1, inst.s2) )
+    meas.register_parameter(inst.m1, setpoints=(inst.s1,))
+
+    with meas.run() as datasaver:
+        for outer in range(10):
+            datasaver.add_result((inst.s1, outer),
+                                (inst.m1, inst.m1()))
+            for inner in range(10):
+                datasaver.add_result((inst.s1, outer),
+                                    (inst.s2, inner),
+                                    (inst.m2, inst.m2()))
+
+    dataid = datasaver.run_id
+    plot_by_id(dataid)
+    plot_by_id(dataid, cmap='bone')
+
+
+def test_appropriate_kwargs():
+    _, ax = plt.subplots()
```

Comment:
It would be slightly cleaner if this tests did not depend on pyplot either by mocking the ax object or by rewriting the context manager to not take ax as an argument